stdlib/TOML/src/print.jl

```diff
@@ -5,6 +5,34 @@ import Dates
 import Base: @invokelatest
 import ..isvalid_barekey_char
 
+function print_toml_escaped(io::IO, s::AbstractString)
+    for c::AbstractChar in s
+        if !isvalid(c)
+            error("TOML print: invalid character $(repr(c)) encountered when printing string")
+        end
+        if c == '\b'
+            Base.print(io, '\\', 'b')
+        elseif c == '\t'
+            Base.print(io, '\\', 't')
+        elseif c == '\n'
+            Base.print(io, '\\', 'n')
+        elseif c == '\f'
+            Base.print(io, '\\', 'f')
+        elseif c == '\r'
+            Base.print(io, '\\', 'r')
+        elseif c == '"'
+            Base.print(io, '\\', '"')
+        elseif c == '\\'
+            Base.print(io, "\\", '\\')
+        elseif Base.iscntrl(c)
+            Base.print(io, "\\u")
+            Base.print(io, string(UInt32(c), base=16, pad=4))
+        else
+            Base.print(io, c)
```

Comment:
Should it throw an error for unrecognized escape sequences?

Reply:
Hm, maybe? Could you give an example.

Reply:
See #22800 and #27238 for discussion and examples of analogous issues in the Julia parser and in `unescape_string`.

Reply:
What I meant is more an example Julia string that should error upon writing it to a TOML file.

Reply:
Oh never mind. I confused escaping with unescaping here.